### PR TITLE
PMP: Add missing NP in clip() and split()

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes
 - Fix clipping plugin behavior and texts.
 - Add an option to permit clipping self-intersecting meshes.
 - Add a NP for splitting self-intersecting meshes

### CODE REVIEW

File: Polyhedron/demo/Polyhedron/Plugins/Operations_on_polyhedra/Clip_polyhedron_plugin.cpp [hunk 1]
```diff
@@ -117,6 +127,26 @@ public :
...
+                                         "If the option `keep closed` is checked, the clipped part of each polyhedron will be closed, "
+                                         "if it has a closed contour on the clipping plane. Otherwise, it will be left open."));
+      else
+        QMessageBox::information(dock_widget, QString("Help"),
+                                 QString("Clipping\n"
+                                         "\n"
+                                         "This function allows to clip all the selected polyhedra against a closed polyhedron, bounding a volume. "
+                                         "What is on the outside of the bound volume will be clipped, and what is on the inside will be kept.\n\n"
+                                         "If the option `keep closed` is checked, the clipped part of each polyhedron will be closed, "
+                                         "if it has a closed contour on the clipping polyhedron. Otherwise, it will be left open."));
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Plugins/Operations_on_polyhedra/Clip_polyhedron_plugin.cpp:147]: That is the sort of code where the usage of [C++11 raw string literals](https://abseil.io/tips/64) might help the readability.

File: Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_clip.cpp [hunk 2]
```diff
@@ -53,6 +53,23 @@ void test()
...
+    TriangleMesh tm1, tm2;
+    std::ifstream("data-clip/clipper_for_tet.off") >> tm2;
+    std::ifstream("data-clip/tet_with_si_to_clip.off") >> tm1;
+
+    PMP::clip(tm1, tm2,
+              params::throw_on_self_intersection(true),
+              params::do_not_modify(true));
+    std::vector<TriangleMesh> meshes;
+    PMP::split_connected_components(tm1, meshes, params::all_default());
+    CGAL_assertion(meshes.size() == 2);
```
Comment (sloriot) [inline Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_clip.cpp:67]:
```suggestion
    assert(meshes.size() == 2);
```

File: Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_clip.cpp [hunk 3]
```diff
@@ -53,6 +53,23 @@ void test()
...
+    //if the order is not deterministc, put the num_vertices in a list and check
+    //if the list does contain all those numbers.
+    CGAL_assertion(num_vertices(meshes[0]) == 21);
```
Comment (sloriot) [inline Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_clip.cpp:70]:
```suggestion
    assert(num_vertices(meshes[0]) == 21);
```

Comment (sloriot) [inline Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_clip.cpp:70]:
```suggestion
    assert(num_vertices(meshes[0]) == 21);
```

File: Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_clip.cpp [hunk 4]
```diff
@@ -120,6 +137,23 @@ void test()
...
   }
 
+  //test with SI
+  {
+    TriangleMesh tm1;
+    std::ifstream("data-clip/tet_with_si_to_clip.off") >> tm1;
+    if(num_vertices(tm1) == 0)
+    {
+      std::cerr<<"File not found. Aborting."<<std::endl;
+      CGAL_assertion(false);
```
Comment (sloriot) [inline Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_clip.cpp:147]:
```suggestion
      assert(false);
```

File: Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_clip.cpp [hunk 5]
```diff
@@ -120,6 +137,23 @@ void test()
...
+      return ;
+    }
+
+    PMP::clip(tm1, K::Plane_3(0,0,1,-4.5),
+               params::throw_on_self_intersection(true)
+               .allow_self_intersections(true));
+    CGAL_assertion(vertices(tm1).size() == 16);
```
Comment (sloriot) [inline Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_clip.cpp:154]:
```suggestion
    assert(vertices(tm1).size() == 16);
```

File: Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_clip.cpp [hunk 6]
```diff
@@ -493,6 +527,29 @@ void test_split_plane()
...
+  }
+
+  PMP::split(tm1, K::Plane_3(0,0,1,-0.5),
+             params::throw_on_self_intersection(true)
+             .allow_self_intersections(true));
+  PMP::split_connected_components(tm1, meshes, params::all_default());
+  CGAL_assertion(meshes.size() == 2);
+  //if the order is not deterministc, put the num_vertices in a list and check
+  //if the list does contain all those numbers.
+  CGAL_assertion(num_vertices(meshes[0]) == 16);
```
Comment (sloriot) [inline Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_clip.cpp:548]:
```suggestion
  assert(num_vertices(meshes[0]) == 16);
```

File: Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_clip.cpp [hunk 7]
```diff
@@ -493,6 +527,29 @@ void test_split_plane()
...
+    std::cerr<<"File not found. Aborting."<<std::endl;
+    CGAL_assertion(false);
+    return ;
+  }
+
+  PMP::split(tm1, K::Plane_3(0,0,1,-0.5),
+             params::throw_on_self_intersection(true)
+             .allow_self_intersections(true));
+  PMP::split_connected_components(tm1, meshes, params::all_default());
+  CGAL_assertion(meshes.size() == 2);
```
Comment (sloriot) [inline Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_clip.cpp:545]:
```suggestion
  assert(meshes.size() == 2);
```

File: Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_clip.cpp [hunk 8]
```diff
@@ -493,6 +527,29 @@ void test_split_plane()
...
 
   CGAL::clear(tm1);
+  meshes.clear();
+
+  //test with SI
+  std::ifstream("data-clip/tet_si_to_split.off") >> tm1;
+  if(num_vertices(tm1) == 0)
+  {
+    std::cerr<<"File not found. Aborting."<<std::endl;
+    CGAL_assertion(false);
```
Comment (sloriot) [inline Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_clip.cpp:537]:
```suggestion
    assert(false);
```

File: Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_clip.cpp [hunk 9]
```diff
@@ -493,6 +527,29 @@ void test_split_plane()
...
+
+  PMP::split(tm1, K::Plane_3(0,0,1,-0.5),
+             params::throw_on_self_intersection(true)
+             .allow_self_intersections(true));
+  PMP::split_connected_components(tm1, meshes, params::all_default());
+  CGAL_assertion(meshes.size() == 2);
+  //if the order is not deterministc, put the num_vertices in a list and check
+  //if the list does contain all those numbers.
+  CGAL_assertion(num_vertices(meshes[0]) == 16);
+  CGAL_assertion(num_vertices(meshes[1]) == 16);
```
Comment (sloriot) [inline Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_clip.cpp:549]:
```suggestion
  assert(num_vertices(meshes[1]) == 16);
```

Comment (sloriot) [reply Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_clip.cpp:549]: etc

**Review by sloriot — APPROVED** [review]

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.2-Ic-46.shtml

**Pull request merged**